### DISCUSSION TITLE
Why is Bloop export and BSP disabled for all -bootstrapped projects?

### DISCUSSION THREAD

**Discussion by ckipp01** [scala/scala3#15854] [Compiler internals Q&A]
I asked this in #scala-contributors on Discord a while back but got no response.

Recently I wanted to work on some stuff in Scaladocs, and when trying to import the project with Metals I noticed that it was including the `commonBootstrappedSettings` which ensures Bloop doesn't export for that target and also has `bspEnabled := false` meaning that using sbt as a build server won't work either.

Having these settings for some projects, like scaladoc, make it very difficult for someone trying to contribute as they get no editor help out of the box. Is this necessary for this module? Why was this done? I see it was done in https://github.com/lampepfl/dotty/pull/11535/commits/442f600bdaa8aecadf677b283ec57ca6927c7e4e, but that commit holds no details at all about why.

**Comment by SethTisue** [comment]
@lrytz do you remember the context?

**Reply by lrytz** [reply]
If I remember correctly this was done to avoid having every module twice in IntelliJ (compiler and compiler-bootstrapped). I assume I missed the fact that scaladoc only exists -bootstrapped, is that the case?

**Reply by ckipp01** [reply]
>  I assume I missed the fact that scaladoc only exists -bootstrapped, is that the case?

I'm not sure. @pikinier20 do you think it'd be possible to get some contributing docs set up for Scaladocs with instructions that might work for Metals users? I see that you have this here: https://github.com/lampepfl/dotty/tree/main/scaladoc#developing, but it makes it sound like Metals will work out of the box.  I tried moving some stuff around to get Scaladoc exported, but then other stuff blew up, and I sort of stopped my dive there.

**Reply by sjrd** [reply]
There's another problem: a -bootstrapped artifact uses as compiler the compiler being built in the build. That is, almost by definition, a compiler not supported by Metals. Not to mention that it can constantly change under Metals' feet. I think that is the bigger issue with the -bootstrapped artifacts in the dotty build.
